build.gradle

```diff
@@ -16,14 +16,57 @@ versions {
 
 ext.isCiServer = System.getenv().get("CI") != null
 
-test {
+tasks.withType(Test) {
     retry {
         maxRetries = isCiServer ? 1 : 0
         maxFailures = 10
         failOnPassedAfterRetry = true
     }
 }
 
+def testJni = tasks.register("testJni", Test) {
+    // See https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/clopts002.html
+    it.jvmArgs "-Xcheck:jni"
+
+    // Only run tests that have the category
+    it.useJUnit {
+        includeCategories 'net.rubygrapefruit.platform.testfixture.JniChecksEnabled'
+    }
+
+    // Check standard output for JNI warnings and fail if we find anything
+    String currentTest = null
+    it.addTestListener(new TestListener() {
+        @Override
+        void beforeSuite(TestDescriptor testDescriptor) {}
+
+        @Override
+        void afterSuite(TestDescriptor testDescriptor, TestResult testResult) {}
+
+        @Override
+        void beforeTest(TestDescriptor testDescriptor) {
+            currentTest = "${testDescriptor.className}.${testDescriptor.displayName}"
+        }
+
+        @Override
+        void afterTest(TestDescriptor testDescriptor, TestResult testResult) {
+            currentTest = null
+        }
+    })
+    List<String> warningsDetected = []
+    it.logging.addStandardOutputListener({ String message ->
+        if (currentTest != null && message.startsWith("WARNING")) {
```

Comment:
So we don't detect any warnings from setup methods and the like?

Reply:
Those methods are executed as part of the tests, so we do catch problems emitted via them.

Reply:
IIRC, `beforeTest` fires after the `setup` method. That is the reason why we didn't have the class name/method name in some of the test directories.

It should be mostly fine, though I think we won't catch any warnings from loading native platform.

Reply:
I tested this and messages in `setup()` were reported.

Reply:
All good then.

Reply:
I think what you mean is `beforeSpec()`, that is not included indeed.

Reply:
...but we don't use it in our tests.